controllers/admin/AdminBlockListingController.php

```diff
@@ -0,0 +1,205 @@
+<?php
+/**
+* 2007-2019 PrestaShop
+*
+* NOTICE OF LICENSE
+*
+* This source file is subject to the Academic Free License (AFL 3.0)
+* that is bundled with this package in the file LICENSE.txt.
+* It is also available through the world-wide-web at this URL:
+* http://opensource.org/licenses/afl-3.0.php
+* If you did not receive a copy of the license and are unable to
+* obtain it through the world-wide-web, please send an email
+* to license@prestashop.com so we can send you a copy immediately.
+*
+* DISCLAIMER
+*
+* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+* versions in the future. If you wish to customize PrestaShop for your
+* needs please refer to http://www.prestashop.com for more information.
+*
+*  @author PrestaShop SA <contact@prestashop.com>
+*  @copyright  2007-2019 PrestaShop SA
+*  @license    http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+*  International Registered Trademark & Property of PrestaShop SA
+*/
+
+class AdminBlockListingController extends ModuleAdminController
+{
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Update the status of a block
+     *
+     * @return string
+     */
+    public function ajaxProcessChangeBlockStatus()
+    {
+        $dt = new DateTime();
+        $PsreassuranceId = (int) Tools::getValue('idpsr');
+        $Status = ((int) Tools::getValue('status') == 1) ? 0 : 1;
+        $newStatus = ($Status == 1) ? 1 : 0;
+
+        $UpdatedDatas = array(
+            'status' => $newStatus,
+            'date_upd' => $dt->format('Y-m-d H:i:s'),
+        );
+        $sWhere = 'id_psreassurance = '.$PsreassuranceId;
+
+        $bStatusChanged = Db::getInstance()->update('psreassurance', $UpdatedDatas, $sWhere);
+
+        $this->ajaxDie($bStatusChanged);
+    }
+
+    /**
+     * ajaxProcessSavePositionByHook
+     *
+     * @return string
+     */
+    public function ajaxProcessSavePositionByHook()
+    {
+        $hook = Tools::getValue('hook');
+        $value = Tools::getValue('value');
+
+        Configuration::updateValue($hook, $value);
+
+        $this->ajaxDie(json_encode('success'));
+    }
+
+    /**
+     * ajaxProcessSaveColor
+     *
+     * @return string
+     */
+    public function ajaxProcessSaveColor()
+    {
+        $color1 = Tools::getValue('color1');
+        $color2 = Tools::getValue('color2');
+
+        Configuration::updateValue('PSR_ICON_COLOR', $color1);
+        Configuration::updateValue('PSR_TEXT_COLOR', $color2);
+
+        $this->ajaxDie(json_encode('success'));
+    }
+
+    /**
+     * ajaxProcessSaveBlockContent
+     *
+     * @return bool|string
+     */
+    public function ajaxProcessSaveBlockContent()
```

Comment:
This function is too long, it's hard to understand what it does and see the different sections.
Consider splitting it in smaller functions, example:
```
public function ajaxProcessSaveBlockContent()
{
    $this->handleBlockValues();
    $this->handlePicto();
    $this->handleUpload();
}
```